gradle-sls-packaging/src/main/java/com/palantir/gradle/dist/service/JavaServiceDistributionPlugin.java

```diff
@@ -254,6 +258,19 @@ public void execute(Task _task) {
             task.setGroup(JavaServiceDistributionPlugin.GROUP_NAME);
             task.setDescription("Runs the specified project using configured mainClass and with default args.");
             task.dependsOn("jar");
+            task.dependsOn(javaAgentConfiguration);
+            task.jvmArgs(ImmutableList.<String>builder()
+                    .addAll(distributionExtension.getDefaultJvmOpts().get())
+                    .addAll(distributionExtension.getGcJvmOptions().get())
+                    .build());
+            task.getJvmArgumentProviders().add(new CommandLineArgumentProvider() {
```

Comment:
Hmm, I didn't know you could use `jvmArgs` and `getJvmArgumentProviders` together - I thought they overwrote each other (perhaps they would if they were declared with `jvmArgs` last?)

Reply:
huh, I'll update the test to verify gc args are present

Reply:
hmm, this is the run task, not sure we test it. Might as well combine the two into a single jCommandLineArgumentProvider